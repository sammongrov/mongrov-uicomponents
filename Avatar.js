import React, { Component } from 'react';
import { View, TouchableOpacity } from 'react-native';
import FastImage from 'react-native-fast-image';
import { styles } from 'react-native-theme';
import PropTypes from 'prop-types';
import AppUtils from '@mongrov/utils';
import Text from './Text';

export default class Avatar extends Component {
  state = {
    urlErr: false,
    loading: false,
  };

  renderAvatar() {
    const { avatarUrl, avatarName, avatarSize } = this.props;
    const { urlErr, loading } = this.state;

    if (avatarUrl && !urlErr && !loading) {
      return (
        <FastImage
          style={{
            width: avatarSize,
            height: avatarSize,
            borderRadius: avatarSize / 2,
          }}
          source={{
            uri: avatarUrl,
            priority: FastImage.priority.high,
          }}
          onError={() => this.setState({ urlErr: true })}
        />
      );
    }
    return <Text style={styles.avatarText}>{AppUtils.avatarInitials(avatarName)}</Text>;
  }

  render() {
    const { statusColor, avatarSize, avatarName, onAvatarPress } = this.props;
    return (
      <View
        style={[
          styles.rowDirection,
          {
            position: 'relative',
          },
        ]}
      >
        <TouchableOpacity
          onPress={onAvatarPress}
          style={[
            styles.avatarImageContainer,
            {
              width: avatarSize,
              height: avatarSize,
              borderRadius: avatarSize / 2,
              backgroundColor: AppUtils.getAvatarColor(avatarName),
            },
          ]}
        >
          {this.renderAvatar()}
        </TouchableOpacity>
        {statusColor !== null && (
          <View
            style={[
              styles.avatarStatusDot,
              {
                backgroundColor: statusColor,
                position: 'absolute',
                right: 0,
              },
            ]}
          />
        )}
      </View>
    );
  }
}

Avatar.propTypes = {
  statusColor: PropTypes.string,
  avatarUrl: PropTypes.string,
  avatarName: PropTypes.string,
  avatarSize: PropTypes.number,
  onAvatarPress: PropTypes.func,
};

Avatar.defaultProps = {
  statusColor: null,
  avatarUrl: null,
  avatarName: '',
  avatarSize: 60,
  onAvatarPress: () => {},
};
