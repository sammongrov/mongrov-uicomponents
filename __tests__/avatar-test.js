// import React from 'react';
// import renderer from 'react-test-renderer';
// // import { Alert } from 'react-native';
// import Avatar from '../Avatar';

// it('Avatar renders correctly', () => {
//   const tree = renderer.create(<Avatar statusColor="red" avatarUrl="url" />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Alert, Image } from 'react-native';
import AppUtils from '@mongrov/utils';
import Avatar from '../Avatar';

const mockMath = global.Math;
mockMath.random = () => 0.5;
global.Math = mockMath;

configure({ adapter: new Adapter() });
const avatarInitials = () => Alert.alert('message');

it('Avatar renders correctly', () => {
  const tree = renderer.create(<Avatar avatarBgColor={AppUtils.getRandomColor()} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('messageCount Component using shallow renders correctly', () => {
  const wrapper = shallow(<Avatar avatarUrl="url" avatarInitials={avatarInitials} />);
  expect(wrapper.instance().props.avatarInitials).toBeTruthy();
});

test('onError props test', () => {
  const wrapper = shallow(<Avatar avatarUrl="url" avatarInitials={avatarInitials} />);
  wrapper
    .find(Image)
    .first()
    .props()
    .onError();
  expect(wrapper.instance().state.urlErr).toBeTruthy();
});

test('statusColor props test', () => {
  const wrapper = shallow(<Avatar statusColor="green" />);
  expect(wrapper.instance().props.statusColor).not.toBeNull();
});
