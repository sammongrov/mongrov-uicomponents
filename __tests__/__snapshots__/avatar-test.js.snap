// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "position": "relative",
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#2E88FF",
        "borderRadius": 30,
        "height": 60,
        "opacity": 1,
        "width": 60,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "OpenSans-Regular",
          },
          null,
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
